### ISSUE TITLE
"Bad Gateway" NGINX error

### ISSUE THREAD

**Issue by solus-hq** [openspace42/aenigma#50]
This is what I'm receiving after trying to install on Ubuntu 18.04 LTS on 3 different cloud VPS providers...

```
502 Bad Gateway
nginx/1.14.2
```

**Comment by rowie** [comment]
can you post the output from the install. script?

**Comment by solus-hq** [comment]
> can you post the output from the install. script?

```
Finished installing ejabberd.

Now setting permissions...

Finished setting permissions.

Now wiping ejabberd to prepare for the new installation...

This is your final warning: if you're not absolutely sure your data is safe, abort the installation now!

| - | Counting down to full ejabberd wipe on this machine... [for 00 more seconds]

Now finally starting up ejabberd...

| - | Now allowing ejabberd to start up correctly... [for 00 more seconds]

The node 'ejabberd@ae01.xmpp.tf' is started with status: started
ejabberd 18.12.1-1~ubuntu18.04 is running in that node

Finished starting ejabberd back up.

Now adding job to daily crontab via cronic...

Finished adding job to daily crontab via cronic.

Now registering ejabberd admin user [if it doesn't yet exist]...

Finished registering ejabberd admin user.

Now updating ejabberd's module repo...

Finished updating ejabberd's module repo...

Now creating 'ADMIN:EVERYBODY' and 'ADMIN:ONLINE' shared roster groups [admin@xmpp.tf can see 'everybody' and 'all online']...

Finished creating shared roster groups.

All | SSLH | operations completed successfully.

All | NGINX | operations completed successfully.

Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

All | APT | operations completed successfully.

You can finally log in to your web administration panel:

 | https://ae01.xmpp.tf |

 | admin@xmpp.tf |

 | PASSWORD HERE |

Press enter to continue...
```

**Comment by rowie** [comment]
and the dns checks are fine?

**Comment by solus-hq** [comment]
Yes, except IPv6, which is not configured. IPv4 checks went just fine. 

The final part of the installer log is: (after generating DH)

```

.........................................................++*++*

Now restarting ejabberd...

Installation complete.

Now removing installer directory...

Run | git clone https://github.com/openspace42/aenigma | once again to download the latest installer.

Exiting installer.
```

**Comment by rowie** [comment]
did you use the latest version?

**Comment by solus-hq** [comment]
Well, I did exactly what the installation manual says, git cloned into current repository and ran `bash aenigma/setup`

nginx is up and running:

```
root@server-1130747-1:/etc/nginx/conf.d# service nginx status
● nginx.service - nginx - high performance web server
   Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
   Active: active (running) since Fri 2019-04-05 07:22:28 UTC; 3min 49s ago
     Docs: http://nginx.org/en/docs/
  Process: 4460 ExecStop=/bin/kill -s TERM $MAINPID (code=exited, status=0/SUCCESS)
  Process: 4461 ExecStart=/usr/sbin/nginx -c /etc/nginx/nginx.conf (code=exited, status=0/SUCCESS)
 Main PID: 4462 (nginx)
    Tasks: 2 (limit: 4915)
   CGroup: /system.slice/nginx.service
           ├─4462 nginx: master process /usr/sbin/nginx -c /etc/nginx/nginx.conf
           └─4463 nginx: worker process
```

ejabberd is running as well according to htop

**Comment by rowie** [comment]
and you try to open the side with your browser and the fqdn?
like https://example.com

**Comment by solus-hq** [comment]
> and you try to open the side with your browser and the fqdn?
> like https://example.com

Yes, you may take a look at this screenshot.

<img width="1136" alt="Screen Shot 2019-04-05 at 10 28 54" src="https://user-images.githubusercontent.com/26515805/55611274-bc241f80-578d-11e9-9a72-8f473d2debd6.png">

<img width="427" alt="Screen Shot 2019-04-05 at 10 31 04" src="https://user-images.githubusercontent.com/26515805/55611370-f8f01680-578d-11e9-828e-dbf5c735c3bf.png">

**Comment by rowie** [comment]
hehe, sorry .... to early in the morning. ;-)
i cant test it atm cause all my servers behing nat. 
sorry, you have to wait for the dev.

**Comment by solus-hq** [comment]
> .

Thanks, I hope I can be useful to help with testing the installation process on various cloud providers (I have 4 popular providers available) and eliminate possible drawbacks.

**Comment by rowie** [comment]
and what do you get without /admin  ... or with the port for the admin behind?
which ports are open? 
netstat -tulpn

**Comment by solus-hq** [comment]
```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:42022           0.0.0.0:*               LISTEN      13228/sshd
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      4462/nginx: master
tcp        0      0 0.0.0.0:25              0.0.0.0:*               LISTEN      26730/master
tcp        0      0 127.0.0.1:443           0.0.0.0:*               LISTEN      4462/nginx: master
tcp        0      0 77.221.147.49:443       0.0.0.0:*               LISTEN      3829/sslh
tcp        0      0 77.221.147.49:34141     0.0.0.0:*               LISTEN      4225/beam.smp
tcp6       0      0 :::5280                 :::*                    LISTEN      4225/beam.smp
tcp6       0      0 :::5444                 :::*                    LISTEN      4225/beam.smp
tcp6       0      0 :::5222                 :::*                    LISTEN      4225/beam.smp
tcp6       0      0 :::42022                :::*                    LISTEN      13228/sshd
tcp6       0      0 :::5223                 :::*                    LISTEN      4225/beam.smp
tcp6       0      0 :::4369                 :::*                    LISTEN      1/systemd
tcp6       0      0 :::5269                 :::*                    LISTEN      4225/beam.smp
tcp6       0      0 :::25                   :::*                    LISTEN      26730/master
tcp6       0      0 2a02:440:0:13::403:443  :::*                    LISTEN      3829/sslh
udp        0      0 77.221.147.49:123       0.0.0.0:*                           15283/ntpd
udp        0      0 10.131.5.0:123          0.0.0.0:*                           15283/ntpd
udp        0      0 127.0.0.1:123           0.0.0.0:*                           15283/ntpd
udp        0      0 0.0.0.0:123             0.0.0.0:*                           15283/ntpd
udp6       0      0 fe80::21c:42ff:fe7d:123 :::*                                15283/ntpd
udp6       0      0 2a02:440:0:13::402:123  :::*                                15283/ntpd
udp6       0      0 2a02:440:0:13::403:123  :::*                                15283/ntpd
udp6       0      0 ::2:123                 :::*                                15283/ntpd
udp6       0      0 ::1:123                 :::*                                15283/ntpd
udp6       0      0 :::123                  :::*                                15283/ntpd
```

**Comment by rowie** [comment]
i cant test it atm cause all my servers behind nat firewalls .. :-(

**Comment by Bernd84** [comment]
i have the same problem, xmpp works fine, but administration panel 502 Bad Gateway

**Comment by solus-hq** [comment]
> i have the same problem, xmpp works fine, but administration panel 502 Bad Gateway

Hello! Have you seen this issue as well (is your server affected) ?

https://github.com/openspace42/aenigma/issues/51#issue-429791848

**Comment by rowie** [comment]
can you paste the /etc/sslh/sslh.cfg if exists?

maybe i am wrong but i think its the sslh.

**Comment by Bernd84** [comment]
> > i have the same problem, xmpp works fine, but administration panel 502 Bad Gateway
> 
> Hello! Have you seen this issue as well (is your server affected) ?
> 
> [#51 (comment)](https://github.com/openspace42/aenigma/issues/51#issue-429791848)

yes, same here "Permission denied, please try again" with the new root password

**Comment by solus-hq** [comment]
> > > i have the same problem, xmpp works fine, but administration panel 502 Bad Gateway
> > 
> > 
> > Hello! Have you seen this issue as well (is your server affected) ?
> > [#51 (comment)](https://github.com/openspace42/aenigma/issues/51#issue-429791848)
> 
> yes, same here "Permission denied, please try again" with the new root password

Seems the script is currently unusable =( If @nordurljosahvida could help us I would really appreciate any input regarding this problem.

**Comment by nordurljosahvida** [comment]
Hi! @rowie thanks for all of the support you're providing to users! @solus-hq @Bernd84 can I access your machines to try and figure out what's wrong?

**Comment by solus-hq** [comment]
@nordurljosahvida Yes, I can surely provide access to my VPS, where do I send access credentials? Is your XMPP ID publicly available?

**Comment by nordurljosahvida** [comment]
Yes you can write me via mail and xmpp at my address nz at os dot vu

**Comment by nordurljosahvida** [comment]
@solus-hq @Bernd84 fixed in v0.6.13. I had recently updated the port for the ejabberd webadmin interface but I had only updated it in the config file for ejabberd v19.02, not ejabberd v18.12.

Simply run `aenigma-upgrade` and the correct version will be installed [ @solus-hq you're already good, see my email for a few questions]

Please consider joining our group chat at xmpp:aenigma@xc.os.vu and - if you can - also consider becoming a supporter on [Patreon](https://www.patreon.com/aenigma_xmpp) to help fund the project in the future.

Thanks!
